src/AlgebraicGeometry/ToricVarieties/NormalToricVarieties/constructors.jl

```diff
@@ -178,6 +178,92 @@ end
+
+@doc raw"""
+    Base.:(==)(tv1::NormalToricVariety, tv2::NormalToricVariety)
+
+In OSCAR, the `==` operator is reserved to check if two normal toric varieties are identical,
+meaning their underlying polyhedral fans are the same. However, this check is computationally
+challenging due to several reasons:
+
+- The ray generators might be scaled.
+- The ray generators might be stored in different orders.
+- The maximal (polyhedral) cones of the polyhedral fan might be stored in different orders.
+- If we fall back on polyhedral fan equality, lineality of the cones must also be considered.
+
+To avoid this computational bottleneck, we have specially designed the `==` method.
+It checks if the memory locations of the two objects in question are identical. If 
+so, our `==` method returns `true`. Otherwise, it raise an error, as shown below.
+
+Note that triple equality `===` (i.e. the check of equal the memory locations) is always
+supported for normal toric varieties. We recommend using it.
+
+Finally, if you truly need to check for two normal toric varieties to be (mathematically)
+identical, then we provide an internal method `slow_equal`. This method is not performant
+(and we currently have no intentions in making improvements to its performance). Therefore,
+expect this method to be slow, potentially painfully so.
+
+# Examples
+```jldoctest
+julia> p2 = projective_space(NormalToricVariety, 2)
+Normal toric variety
+
+julia> f2 = hirzebruch_surface(NormalToricVariety, 2)
+Normal toric variety
+
+julia> p2 === f2
+false
+
+julia> p2 === p2
+true
+
+julia> p2 == f2
+ERROR: Equality of normal toric varieties is computationally very demanding. More details in the documentation.
+[...]
+```
+"""
+function Base.:(==)(tv1::NormalToricVariety, tv2::NormalToricVariety)
+  tv1 === tv2 && return true
+  error("Equality of normal toric varieties is computationally very demanding. More details in the documentation.")
+end
+
+
+@doc raw"""
+    slow_equal(tv1::NormalToricVariety, tv2::NormalToricVariety)
+
+This methods return `true` if the toric varieties are identical (as mathematical objects)
+and otherwise returns `false`.
+
+Note that this function is not very performant (and we currently have no intentions to
+improve on this), as the conducted tests are computationally fairly involved. Therefore,
+expect this method to be slow, potentially painfully so. Also, compare our design of the
+`==` method for normal toric varieties.
+
+# Examples
+```jldoctest
+julia> p2 = projective_space(NormalToricVariety, 2)
+Normal toric variety
+
+julia> f2 = hirzebruch_surface(NormalToricVariety, 2)
+Normal toric variety
+
+julia> p2 == f2
+ERROR: Equality of normal toric varieties is computationally very demanding. More details in the documentation.
+[...]
+
+julia> Oscar.slow_equal(p2, f2)
+false
+```
+"""
+function slow_equal(tv1::NormalToricVariety, tv2::NormalToricVariety)
+  return Set(maximal_cones(tv1)) == Set(maximal_cones(tv2))
+end
```

Comment:
Personally I would just recommend to leave out this oneliner, since everyone needing the check can do that by themselves and verify that it is a bad idea. However, if you insist, you should prepend some easy checks:
```suggestion
function slow_equal(tv1::NormalToricVariety, tv2::NormalToricVariety)
  tv1 === tv2 && return true
  ambient_dim(tv1) == ambient_dim(tv2) || return false
  f_vector(tv1) == f_vector(tv2) || return false
  return Set(maximal_cones(tv1)) == Set(maximal_cones(tv2))
end
```

Reply:
One other option would be to hijack the method `issetequal` (`is_set_equal`). It is a bit more descriptive compared to `slow_equal`. Just some backseat-programming, feel free to ignore it.

Reply:
Along your suggestion, I have removed the implementation of this function from OSCAR. Still, since we have the blueprint now and this might be useful to people, I have added this as code block to the documentation.